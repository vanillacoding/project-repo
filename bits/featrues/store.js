import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { userSlice } from './userSlice';
import { habitSlice } from './habitSlice';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistStore, persistReducer } from 'redux-persist';

const userPersistConfig = {
  key: 'user',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const habitPersistConfig = {
  key: 'habit',
  storage: AsyncStorage,
  blacklist: ['isSuccess', 'isError', 'isFetching']
};

const reducer = {
  user: persistReducer(userPersistConfig, userSlice.reducer),
  habit: persistReducer(habitPersistConfig, habitSlice.reducer)
};

const middleware = [ReduxThunk, createLogger()];

const store = configureStore({
  reducer,
  middleware
});

const persistedStore = persistStore(store);

export { store, persistedStore };
