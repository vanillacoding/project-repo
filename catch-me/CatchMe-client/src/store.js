import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger"

import single from "./redux/slices/singlePlaySlice"; 
import multiple from "./redux/slices/multiplaySlice";
import socketMiddleware from "./redux/middleware/socket";

const store = configureStore({
  reducer: {
    single,
    multiple,
  },
  middleware: (getDefaultMiddleware) => {
    if (process.env.NODE_ENV !== "production") {
      return getDefaultMiddleware().concat([logger, socketMiddleware()]);
    }

    return getDefaultMiddleware().concat([logger, socketMiddleware()]);
  },
  devTools: process.env.NODE_ENV !== "production",
});

export default store;
