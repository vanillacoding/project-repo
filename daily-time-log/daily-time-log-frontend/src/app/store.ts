import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import logger from "redux-logger";

import authReducer from "../features/auth/authSlice";
import calendarReducer from "../features/calendar/calendarSlice";
import schedulesReducer from "../features/schedules/schedulesSlice";
import milestonesReducer from "../features/milestones/milestonesSlice";
import timeLogReducer from "../features/timeLog/timeLogSlice";
import settingReducer from "../features/setting/settingSlice";

const store = configureStore({
  reducer: {
    auth: authReducer,
    calendar: calendarReducer,
    schedules: schedulesReducer,
    milestones: milestonesReducer,
    timeLog: timeLogReducer,
    setting: settingReducer,
  },
  middleware: (getDefaultMiddleware) => {
    return process.env.REACT_APP_NODE_ENV !== "production"
      ? getDefaultMiddleware().concat(logger)
      : getDefaultMiddleware();
  },
  devTools: process.env.REACT_APP_NODE_ENV !== "production",
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
