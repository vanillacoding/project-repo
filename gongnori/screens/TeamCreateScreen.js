import React, { useMemo, useState } from "react";
import { StyleSheet, View, Text, Image, TextInput } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { StatusBar } from "expo-status-bar";
import { useSelector } from "react-redux";
import PropTypes from "prop-types";

import produce from "immer";
import _ from "lodash";

import DismissKeyboard from "../components/DismissKeyboard";
import DropDown from "../components/DropDown";
import CustomButton from "../components/CustomButton";
import CompletionModal from "../components/CompletionModal";
import InputAlertModal from "../components/InputAlertModal";
import SpinnerLoading from "../components/SpinnerLoading";

import useHeaderRight from "../hooks/useHeaderRight";
import usePickImage from "../hooks/usePickImage";

import * as colors from "../constants/colors";
import * as fonts from "../constants/fonts";
import * as sizes from "../constants/sizes";

const DEFAULT_EMBLEM = "https://minho-bucket.s3.ap-northeast-2.amazonaws.com/blank_profile.png";

export default function MatchCreateScreen({ navigation }) {
  const [isHeaderRightLoading, isCompletionShown, isInputInvalid] = useSelector((state) => {
    return [
      state.loading.isHeaderRightLoading,
      state.loading.isCompletionShown,
      state.loading.isInputInvalid,
    ];
  }, (prev, next) => _.cloneDeep(prev) === _.cloneDeep(next));

  const [team, setTeam] = useState({
    name: null,
    location: null,
    sports: null,
  });

  const locations = useSelector((state) => {
    return state.user.locations;
  }, (prev, next) => _.cloneDeep(prev) === _.cloneDeep(next));

  const locationOptions = useMemo(() => {
    return locations.map((location) => {
      return `${location.city} ${location.district}`;
    });
  }, []);

  const sports = useSelector((state) => {
    return state.app.sports;
  }, (prev, next) => _.cloneDeep(prev) === _.cloneDeep(next));

  const sportsOptions = useMemo(() => {
    return sports.map((item) => item.koreanName);
  }, []);

  const handleSelectSports = (index) => {
    setTeam(produce(team, (draft) => {
      draft.sports = sports[index];
    }));
  };

  const handleSelectLocation = (index) => {
    setTeam(produce(team, (draft) => {
      draft.location = locations[index];
    }));
  };

  const handleChangeName = (value) => {
    setTeam(produce(team, (draft) => {
      draft.name = value;
    }));
  };

  const [image, imageS3, pickImage] = usePickImage(DEFAULT_EMBLEM);

  useHeaderRight(
    { navigation, title: "만들기" },
    { method: "POST", path: "team", data: { ...team, imageS3 } }
  );

  return (
    <DismissKeyboard>
      <SafeAreaView style={styles.container}>
        <StatusBar backgroundColor="white" barStyle="light-content" />
        <SpinnerLoading
          visible={isHeaderRightLoading}
          content={"Team Creating"}
        />
        <InputAlertModal
          content={"팀 정보를 입력해주세요."}
          visible={isInputInvalid}
        />
        <CompletionModal
          content={"팀을 만들었습니다."}
          visible={isCompletionShown}
        />
        <View style={styles.emblem}>
          <View style={styles.imageBox}>
            <Image
              style={styles.image}
              source={{ uri: image }}
            />
          </View>
          <CustomButton
            title={"앨범에서 선택"}
            style={styles.button}
            onPress={pickImage}
          />
        </View>
        <View style={styles.inputContainer}>
          <View style={styles.input}>
            <Text style={styles.title}>{"팀명"}</Text>
            <TextInput
              value={team.name}
              style={styles.field}
              placeholder={"팀 이름을 입력하세요."}
              autoCompleteType="off"
              onChangeText={handleChangeName}
            />
          </View>
          <View style={styles.input}>
            <Text style={styles.title}>종목</Text>
            <DropDown
              defaultValue={"종목을 선택하세요."}
              options={sportsOptions}
              style={styles.field}
              onSelect={handleSelectSports}
            />
          </View>
          <View style={styles.input}>
            <Text style={styles.title}>지역</Text>
            <DropDown
              defaultValue={"동네를 선택하세요."}
              options={locationOptions}
              style={styles.field}
              onSelect={handleSelectLocation}
            />
          </View>
        </View>
      </SafeAreaView>
    </DismissKeyboard>
  );
}

MatchCreateScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    minHeight: sizes.DEVICE_HEIGHT,
    alignItems: "center",
    backgroundColor: colors.SECONDARY_GRAY,
  },
  inputContainer: {
    justifyContent: "space-around",
    width: 0.9 * sizes.DEVICE_WIDTH,
    height: 0.3 * sizes.DEVICE_HEIGHT,
    marginVertical: 30,
    borderRadius: 5,
    backgroundColor: colors.PRIMARY_YELLOW,
    elevation: 5,
    shadowColor: "rgb(50, 50, 50)",
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      height: -1,
      width: 0,
    },
  },
  input: {
    flexDirection: "row",
    alignItems: "center",
    height: 0.1 * sizes.DEVICE_HEIGHT,
  },
  title: {
    height: 0.05 * sizes.DEVICE_HEIGHT,
    marginHorizontal: 15,
    fontSize: sizes.TERTIARY_FONT_SIZE,
    fontFamily: fonts.NOTO_SANS_KR_400_REGULAR,
    textAlign: "left",
    textAlignVertical: "center",
    includeFontPadding: false,
  },
  field: {
    width: 0.4 * sizes.DEVICE_WIDTH,
    height: 0.04 * sizes.DEVICE_HEIGHT,
    borderRadius: 5,
    backgroundColor: colors.SECONDARY_WHITE,
    fontSize: sizes.QUATERNARY_FONT_SIZE,
    fontFamily: fonts.NOTO_SANS_KR_300_LIGHT,
    textAlignVertical: "center",
    textAlign: "center",
    includeFontPadding: false,
  },
  emblem: {
    alignItems: "center",
    marginTop: 30,
  },
  imageBox: {
    justifyContent: "center",
    alignItems: "center",
    width: 0.3 * sizes.DEVICE_WIDTH,
    height: 0.15 * sizes.DEVICE_HEIGHT,
    marginBottom: 20,
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "contain",
  },
  button: {
    width: 0.3 * sizes.DEVICE_WIDTH,
    height: 0.1 * sizes.DEVICE_WIDTH,
    borderRadius: 15,
    backgroundColor: colors.PRIMARY_BLUE,
    color: colors.SECONDARY_WHITE,
    fontSize: sizes.TERTIARY_FONT_SIZE,
  },
});
