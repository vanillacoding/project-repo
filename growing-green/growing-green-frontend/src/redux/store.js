import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer from './root-reducer';

let middleware = [thunk];

if (process.env.NODE_ENV === 'development') {
  middleware = [...middleware, logger];
}

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export default store;
