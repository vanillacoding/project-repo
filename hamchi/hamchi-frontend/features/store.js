import { configureStore } from '@reduxjs/toolkit';
import ReduxThunk from 'redux-thunk';
import { userSlice } from './userSlice';
import { postSlice } from './postSlice';
import { chatSlice } from './chatSlice';
import { filteredPostSlice } from './filteredPostSlice';
import { createLogger } from 'redux-logger';

const middleware = [ReduxThunk];

const store = configureStore({
  reducer:
  {
    user: userSlice.reducer,
    post: postSlice.reducer,
    chat: chatSlice.reducer,
    filteredPost: filteredPostSlice.reducer
  },
  middleware: middleware
});

export default store;
