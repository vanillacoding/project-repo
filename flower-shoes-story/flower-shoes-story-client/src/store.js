import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import storage from "redux-persist/lib/storage";

import userReducer from "./features/userSlice";
import preloadReducer from "./features/preloadSlice";

const reducers = combineReducers({
  user: userReducer,
  preload: preloadReducer,
});

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["user"],
};

const persistedReducer = persistReducer(persistConfig, reducers);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [logger],
});

export default store;
