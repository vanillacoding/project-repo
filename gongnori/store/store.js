import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { createLogger } from "redux-logger";
import { NODE_ENV } from "@env";
import reducer from "../reducers/reducer";

const middleware = [thunk];

if (NODE_ENV === "development") {
  middleware.push(createLogger());
}

export default store = createStore(reducer, applyMiddleware(...middleware));
