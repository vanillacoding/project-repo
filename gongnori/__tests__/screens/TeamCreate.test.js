import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import TeamCreateScreen from "../../screens/TeamCreateScreen";
import store from "../../store/store";

describe("<TeamCreateScreen />", () => {
  test("snapshot test", () => {
    const mockFn = jest.fn();

    const navigation = {
      "addListener": mockFn,
      "canGoBack": mockFn,
      "dangerouslyGetParent": mockFn,
      "dangerouslyGetState": mockFn,
      "dispatch": mockFn,
      "goBack": mockFn,
      "isFocused": mockFn,
      "navigate": mockFn,
      "pop": mockFn,
      "popToTop": mockFn,
      "push": mockFn,
      "removeListener": mockFn,
      "replace": mockFn,
      "reset": mockFn,
      "setOptions": mockFn,
      "setParams": mockFn,
    };

    const component = renderer.create(
      <Provider store={store}>
        <TeamCreateScreen
          navigation={navigation}
        />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
