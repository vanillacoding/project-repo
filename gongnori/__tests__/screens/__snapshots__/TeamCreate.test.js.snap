// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamCreateScreen /> snapshot test 1`] = `
<RNCSafeAreaView
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBEFF5",
      "flex": 1,
      "minHeight": 1334,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.2)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "NotoSansKR_100Thin",
                  "fontSize": 18,
                },
              ]
            }
          >
            Team Creating
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F6F6F6",
            "borderRadius": 5,
            "fontFamily": "NotoSansKR_500Medium",
            "fontSize": 18,
            "height": 133.4,
            "includeFontPadding": false,
            "justifyContent": "center",
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 375,
          }
        }
      >
        팀 정보를 입력해주세요.
      </Text>
    </View>
  </Modal>
  <Modal
    hardwareAccelerated={false}
    testID="modal"
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="touchable-without-feedback"
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F6F6F6",
            "borderRadius": 5,
            "fontFamily": "NotoSansKR_500Medium",
            "fontSize": 18,
            "height": 133.4,
            "includeFontPadding": false,
            "justifyContent": "center",
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 375,
          }
        }
      >
        팀을 만들었습니다.
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 200.1,
          "justifyContent": "center",
          "marginBottom": 20,
          "width": 225,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://minho-bucket.s3.ap-northeast-2.amazonaws.com/blank_profile.png",
          }
        }
        style={
          Object {
            "height": "100%",
            "resizeMode": "contain",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#283593",
          "borderRadius": 15,
          "height": 75,
          "justifyContent": "center",
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "opacity": 1,
          "width": 225,
        }
      }
      testID="custom-button"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 16,
            "height": "100%",
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": "100%",
          }
        }
      >
        앨범에서 선택
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FDF3A7",
        "borderRadius": 5,
        "elevation": 5,
        "height": 400.2,
        "justifyContent": "space-around",
        "marginVertical": 30,
        "shadowColor": "rgb(50, 50, 50)",
        "shadowOffset": Object {
          "height": -1,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
        "width": 675,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 16,
            "height": 66.7,
            "includeFontPadding": false,
            "marginHorizontal": 15,
            "textAlign": "left",
            "textAlignVertical": "center",
          }
        }
      >
        팀명
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCompleteType="off"
        onChangeText={[Function]}
        placeholder="팀 이름을 입력하세요."
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "fontFamily": "NotoSansKR_300Light",
            "fontSize": 14,
            "height": 53.36,
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value={null}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 16,
            "height": 66.7,
            "includeFontPadding": false,
            "marginHorizontal": 15,
            "textAlign": "left",
            "textAlignVertical": "center",
          }
        }
      >
        종목
      </Text>
      <View
        adjustFrame={[Function]}
        animated={true}
        defaultIndex={-1}
        defaultValue="종목을 선택하세요."
        disabled={false}
        dropdownStyle={
          Object {
            "height": 0,
            "width": 300,
          }
        }
        dropdownTextStyle={
          Object {
            "color": "black",
            "fontFamily": "NotoSansKR_300Light",
            "fontSize": 14,
            "height": 24,
            "includeFontPadding": false,
            "marginVertical": 5,
            "paddingVertical": 0,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
        isFullWidth={false}
        keyboardShouldPersistTaps="never"
        multipleSelect={false}
        onSelect={[Function]}
        options={Array []}
        renderButtonComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        renderRightComponent={[Function]}
        renderRowComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        scrollEnabled={true}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "height": 53.36,
            "justifyContent": "center",
            "width": 300,
          }
        }
        testID="modal-drop-down"
        textStyle={
          Object {
            "color": undefined,
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 14,
            "height": 53.36,
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 300,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "NotoSansKR_400Regular",
                      "fontSize": 14,
                      "height": 53.36,
                      "includeFontPadding": false,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": 300,
                    },
                    undefined,
                  ],
                ]
              }
            >
              종목을 선택하세요.
            </Text>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 16,
            "height": 66.7,
            "includeFontPadding": false,
            "marginHorizontal": 15,
            "textAlign": "left",
            "textAlignVertical": "center",
          }
        }
      >
        지역
      </Text>
      <View
        adjustFrame={[Function]}
        animated={true}
        defaultIndex={-1}
        defaultValue="동네를 선택하세요."
        disabled={false}
        dropdownStyle={
          Object {
            "height": 0,
            "width": 300,
          }
        }
        dropdownTextStyle={
          Object {
            "color": "black",
            "fontFamily": "NotoSansKR_300Light",
            "fontSize": 14,
            "height": 24,
            "includeFontPadding": false,
            "marginVertical": 5,
            "paddingVertical": 0,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
        isFullWidth={false}
        keyboardShouldPersistTaps="never"
        multipleSelect={false}
        onSelect={[Function]}
        options={Array []}
        renderButtonComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        renderRightComponent={[Function]}
        renderRowComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        scrollEnabled={true}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "height": 53.36,
            "justifyContent": "center",
            "width": 300,
          }
        }
        testID="modal-drop-down"
        textStyle={
          Object {
            "color": undefined,
            "fontFamily": "NotoSansKR_400Regular",
            "fontSize": 14,
            "height": 53.36,
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 300,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": undefined,
                      "fontFamily": "NotoSansKR_400Regular",
                      "fontSize": 14,
                      "height": 53.36,
                      "includeFontPadding": false,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": 300,
                    },
                    undefined,
                  ],
                ]
              }
            >
              동네를 선택하세요.
            </Text>
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
